lib/mllp/client.ex

```diff
@@ -464,6 +468,34 @@ defmodule MLLP.Client do
     stop_connection(state, reason, "process terminated")
   end
 
+  defp recv_ack(state, opts, buffer) do
+    case state.tcp.recv(state.socket, 0, opts.reply_timeout) do
+      {:ok, reply} ->
+        new_buf = buffer <> reply
+        check = byte_size(new_buf) - 3
+
+        case new_buf do
+          <<@header, _ack::binary-size(check), @trailer>> ->
+            {:ok, new_buf}
+
+          <<@header, _rest::binary>> ->
+            case :binary.split(new_buf, state.mllp_trailer, [:global]) do
```

Comment:
We use binary split here directly to avoid some overhead with say calling String.split/2. That is, calling `String.split/3`, which in turn will compute the scope for the split, and then finally call `:binary.split/3`. Instead, we just split with a global split which is equivlent to what String.split/3 would do for this case. 

What's more, we pass in pre-compiled binary pattern.

Reply:
Curious, how much performance we are gaining here using `:binary.split` vs `String.split`?

Reply:
Well, we'll be ditching this part, but it actually (at least on my machine) was not enough to justify the optimization (about a microsecond!).